routes/extensibleEnums.js

```diff
@@ -0,0 +1,70 @@
+var express = require('express');
+var router = express.Router();
+var utils = require('../util/utils')
+
+// valid enums
+var tommy = {
+    "DaysOfWeek":"Monday",
+    "IntEnum":"1",
+    "name":"Tommy Tomson"
+};
+
+// unexpected enum
+var casper = {
+    "DaysOfWeek":"Weekend",
+    "IntEnum":"2",
+    "name":"Casper Ghosty"
+};
+
+// enum from allowed values
+var scooby = {
+    "DaysOfWeek":"Thursday",
+    "IntEnum":"2.1",
+    "name":"Scooby Scarface"
+};
+
+// roundtripping test
+var retriever = {
+    "name":"Retriever"
+}
+
+
+var pathitem = function(coverage) {
+    router.get('/pet/:petId', function(req, res, next) {
+        var petId = req.params.petId;
+        console.log('Inside pathItem handler with petId "' + petId +'"\n');
+        if (petId === 'tommy') {
+            coverage['expectedEnum']++;
+            res.status(200).send(JSON.stringify(tommy));
+        } 
+        else if(petId === 'casper'){
+            res.status(200).send(JSON.stringify(casper));
+            coverage['unexpectedEnum']++;
+        }
+        else if(petId === 'scooby'){
+            res.status(200).send(JSON.stringify(scooby));
+            coverage['allowedValueEnum']++;
+        }
+        else{
+            utils.send400(res, next, 'Pet not found for '+petId);
+        }
+    });
+
+    router.post('/pet/addPet', function(req, res, next) {
+        var petName = req.body.name;
+        console.log('Inside addPet for '+petName+'\n');
+        if (petName === 'Retriever') {
```

Comment:
Why the stateful implementation via `retriever`? Why not simply reflect back the request (and maybe check that the request contains DaysOfWeek and IntEnum, to ensure it's not boring)?

Reply:
The other 2 tests cover that, this was with the intention of pure roundtripping.

Reply:
huh? my point is: this test isn't purely round tripping. Is uses some object outside of the function (why?), manipulates it, sends it back - why not literally round-trip?

Reply:
Ahh yes, let me fix that asap